robot-db/src/main/java/repositories/AuthorInput.java

```diff
@@ -0,0 +1,39 @@
+package repositories;
+
+import dao.AuthorDAO;
+import domain.Author;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class tests if Author with given name is in database if not it put him
+ */
+
+@Repository
+public class AuthorInput {
+
+    private Map<String, Author> authors;
+    @Autowired
+    private AuthorDAO authorDAO;
+
+    @PostConstruct
+    private void init () {
+        authors = new HashMap<>();
+        authorDAO.findAll().forEach(author -> authors.put(author.getName(), author));
+    }
+
+
+    Author saveAuthor (String name) {
+        if (authors.containsKey(name)) {
+            return authors.get(name);
+        }
+        Author author = Author.builder().name(name).build();
+        authors.put(name, author);
+        authorDAO.save(author);
+        return author;
```

Comment:
No mention whether cache succeeded or failed. Perhaps you don't care, but it's a bad practice. Obviously you call save in only one case here. So, perhaps logging? To have a trace that it happened? Or that it needn't happen?